src/Domain/Amr/Tags/Flags.hpp

```diff
@@ -0,0 +1,19 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "Domain/Amr/Flag.hpp"
+
+namespace domain::amr::Tags {
+/// domain::amr::Flag%s for an Element.
+template <size_t VolumeDim>
+struct Flags : db::SimpleTag {
```

Comment:
Another questions: Are you currently planning this tag should be up-to-date at the point when AMR is triggered and serve as the interface to the AMR actions? Or is this tag meant for internal use, e.g. we pass an `AmrDriverTag` to the AMR actions which will be queried for the flags at the appropriate time and the flags will then be stored in this tag? I think either could work, but maybe using this tag as the interface is the most flexible. What do you think?

Reply:
In the initial design, these tags are mutated and used by the AMR actions.  I am working on an developer guide overview  to outline the algorithm and will open a draft PR within a day or two which may make things clearer

Reply:
see #3837 for the discussion of block-based AMR